engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/ExternalTaskRestService.java

```diff
@@ -91,4 +91,11 @@
   @Produces(MediaType.APPLICATION_JSON)
   BatchDto setRetriesAsync(SetRetriesForExternalTasksDto retriesDto);
 
+  @GET
+  @Path("/topic-names")
+  @Produces(MediaType.APPLICATION_JSON)
+  List<String> getTopicNames(@QueryParam("withLockedTasks") boolean withLockedTasks,
```

Comment:
Let's use a dedicated DTO instead of `List<String>` to remain consistent with all other existing REST API endpoints:
* Introduce a new DTO class (e. g. `ExternalTaskTopicNameDto`) located under `org.camunda.bpm.engine.rest.dto.externaltask`
  * The class should have the attribute `topicName` of type `String`
  * Introduce a getter & setter for the attribute
* Convert the list of strings to a list of `ExternalTaskTopicNameDto`s
* Change the method signature to `List<ExternalTaskTopicNameDto>`

Reply:
@tasso94, do we really want that indirection in the response?

Right now the response would look like this: `[ "foo", "bar", ...]` (basically, a flat list of topic names)

With the requested change, we would introduce another indirection which results in such a response:
```
[
 { "topicName": "foo" },
 { "topicName": "bar" },
 { "topicName": ... }
]
```

As a user, I need to resolve that indirection in my code.

Reply:
I agree with you: a flat list is nicer to handle. However, this would be the very first endpoint where we introduce such a flat list. For instance, the process engine endpoint has this indirection as well: https://docs.camunda.org/manual/7.12/reference/rest/engine/get-names/

Reply:
Let's keep it as is since it's easier for users.